nncf/torch/pruning/filter_pruning/algo.py

```diff
@@ -668,66 +666,13 @@ def _apply_binary_mask_to_module_weight_and_bias(module: torch.nn.Module,
-        """
-        Creating a table with comparison of model params number before and after pruning.
-
-        :param old_modules_info: Information about pruned layers before actually prune layers.
-        :param new_modules_info: Information about pruned layers after actually prune layers.
-        """
-        table = Texttable()
-        header = ["Name", "Weight's shape", "New weight's shape", "Bias shape", "New bias shape",
-                  "Weight's params count", "New weight's params count",
-                  "Mask zero %", "Layer PR"]
-        data = [header]
-
-        for layer in old_modules_info.keys():
-            assert layer in new_modules_info
-
-            drow = {h: 0 for h in header}
-            drow["Name"] = layer
-            drow["Weight's shape"] = old_modules_info[layer]['w_shape']
-            drow["New weight's shape"] = new_modules_info[layer]['w_shape']
-            drow["Bias shape"] = old_modules_info[layer]['b_shape']
-            drow["New bias shape"] = new_modules_info[layer]['b_shape']
-
-            drow["Weight's params count"] = old_modules_info[layer]['params_count']
-            drow["New weight's params count"] = new_modules_info[layer]['params_count']
-
-            drow["Mask zero %"] = old_modules_info[layer]['mask_pr']
-
-            drow["Layer PR"] = 1 - new_modules_info[layer]['params_count'] / old_modules_info[layer]['params_count']
-            row = [drow[h] for h in header]
-            data.append(row)
-        table.add_rows(data)
-        return table
-
     def prepare_for_export(self):
         """
-        This function discards the pruned filters based on the binary masks
-        before exporting the model to ONNX.
+        Applies pruning masks to layer weights before exporting the model to ONNX.
         """
         self._apply_masks()
-        model = self._model.eval().cpu()
-        graph = model.get_original_graph()
-
-        parameters_count_before = model.get_parameters_count_in_model()
-        flops = model.get_MACs_in_model()
-        pruned_layers_stats = self.get_stats_for_pruned_modules()
-
-        init_output_masks_in_graph(graph, self.pruned_module_groups_info.get_all_nodes())
-        model_pruner = ModelPruner(model, graph, PT_PRUNING_OPERATOR_METATYPES)
-        model_pruner.prune_model()
-
-        parameters_count_after = model.get_parameters_count_in_model()
-        flops_after = model.get_MACs_in_model()
-        new_pruned_layers_stats = self.get_stats_for_pruned_modules()
-        stats = self.create_stats_table_for_pruning_export(pruned_layers_stats, new_pruned_layers_stats)
-
-        nncf_logger.info(stats.draw())
-        nncf_logger.info('Final Model Pruning Rate = %.3f', 1 - parameters_count_after / parameters_count_before)
-        nncf_logger.info('Total MAC pruning level = %.3f', 1 - flops_after / flops)
+        stats = self.statistics()
+        nncf_logger.info(stats.to_str())
```

Comment:
Could you please provide some screenshots of how a sample output will be changed after these changes?

Reply:
Thanks for your update!

I would suggest not log any statistics in the export_model function, but I think it will be useful for users to log all statistics by model before export. It can be done via changing examples. What do you think?